lab_5_scrapper/scrapper.py

```diff
@@ -13,70 +73,116 @@ def __init__(self, path_to_config: Path) -> None:
+        headers = config_data.get('headers')
+        total_articles_to_find_and_parse = config_data.get('total_articles_to_find_and_parse')
+        encoding = config_data.get('encoding')
+        timeout = config_data.get('timeout')
+        should_verify_certificate = config_data.get('should_should_verify_certificate')
+        headless_mode = config_data.get('headless_mode')
+
+        if not isinstance(seed_urls, list):
+            raise IncorrectSeedURLError
+
+        for seed_url in seed_urls:
+            if not re.match(r'https?://w?w?w?.', seed_url):
+                raise IncorrectSeedURLError
+
+        if not isinstance(total_articles_to_find_and_parse, int) or isinstance(
+                total_articles_to_find_and_parse, bool) \
+                or total_articles_to_find_and_parse < 0:
+            raise IncorrectNumberOfArticlesError
+
+        if total_articles_to_find_and_parse < 1 or total_articles_to_find_and_parse > 150:
+            raise NumberOfArticlesOutOfRangeError
+
+        if not isinstance(headers, dict):
+            raise IncorrectHeadersError
+
+        if not isinstance(encoding, str):
+            raise IncorrectEncodingError
+
+        if not isinstance(timeout, int) or timeout < 1 or timeout > 60:
+            raise IncorrectTimeoutError
+
+        if not isinstance(should_verify_certificate, bool) or not isinstance(headless_mode, bool):
+            raise IncorrectVerifyError
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        return self.config_content.seed_urls
 
     def get_num_articles(self) -> int:
         """
         Retrieve total number of articles to scrape
         """
-        pass
+        return self.config_content.total_articles
 
     def get_headers(self) -> dict[str, str]:
         """
         Retrieve headers to use during requesting
         """
-        pass
+        return self.config_content.headers
 
     def get_encoding(self) -> str:
         """
         Retrieve encoding to use during parsing
         """
-        pass
+        return self.config_content.encoding
 
     def get_timeout(self) -> int:
         """
         Retrieve number of seconds to wait for response
         """
-        pass
+        return self.config_content.timeout
 
     def get_verify_certificate(self) -> bool:
         """
         Retrieve whether to verify certificate
         """
-        pass
+        return self.config_content.should_verify_certificate
 
     def get_headless_mode(self) -> bool:
         """
         Retrieve whether to use headless mode
         """
-        pass
+        return self.config_content.headless_mode
 
 
 def make_request(url: str, config: Config) -> requests.models.Response:
     """
     Delivers a response from a request
     with given configuration
     """
-    pass
+    return requests.get(url, headers=config.get_headers(), timeout=config.get_timeout())
```

Comment:
you also need to set encoding to response

Reply:
done